# heartmiRNA

This repository assembles data and code of the small RNA analyses from Herdman et al. 2020 zebrafish heart development time course.

## Contents

In this repository, you will find all analysis of the ribozero dataset from abundance estimates with kallisto to differential expression analysis with DESeq2.
This work is divided into folders:

* miRge miRNA counting
    * Markdown file outlining our use of miRge software
    * Long-form counts data file
* Differential expression analysis (DESeq2)
    * R script and Markdown file
    * Diagnostic Figures subfolder
    * Figures subfolder
    * Differential expression analysis results table
* DPGP clustering
* miRanda miRNA target prediction
* TargetScanFish miRNA target prediction
* Correlation analysis
* Combined prediction pipeline

## Experimental Materials & Methods

## Bioinformatic Analysis

### Authors

* **Chelsea Herdman** [Github](https://github.com/chelseaherdman)
* **Bradley L. Demarest** [Github](https://github.com/bdemarest)

Please contact [Chelsea Herdman](cherdman@genetics.utah.edu) with any questions or concerns. 

***
### References
* Lu Y, Baras AS, Halushka MK (2018), _miRge 2.0 for comprehensive analysis of microRNA sequencing data_, BMC Bioinformatics, 19, 275. doi:https://doi.org/10.1186/s12859-018-2287-y

***
